addons/purchase_stock/models/account_move_line.py

```diff
@@ -131,7 +131,7 @@ def _generate_price_difference_vals(self, layers):
                     sign = 1
                     layers_to_consume = []
                     for layer in qty_to_invoice_per_layer:
-                        if layer.stock_move_id._is_in():
+                        if layer.stock_move_id._is_in() or layer.stock_move_id._is_dropshipped():
```

Comment:
I'm wondering if this same issue can occur with refunds: https://github.com/odoo/odoo/pull/169806/files#diff-ee09db59fa59c453713a0602a32630c0047aab02de6e8f7bbdeb41bac7092e90R127 .